src/validationinterface.cpp

```diff
@@ -133,47 +137,62 @@ void CMainSignals::UpdatedBlockTip(const CBlockIndex *pindexNew, const CBlockInd
     // the chain actually updates. One way to ensure this is for the caller to invoke this signal
     // in the same critical section where the chain is updated
 
-    m_internals->m_schedulerClient.AddToProcessQueue([pindexNew, pindexFork, fInitialDownload, this] {
+    auto func = __func__;
+    m_internals->m_schedulerClient.AddToProcessQueue([func, pindexNew, pindexFork, fInitialDownload, this] {
+        if (pindexFork) LogPrint(BCLog::VALIDATIONINTERFACE, "%s: fork block hash=%s\n", func, pindexFork->GetBlockHash().ToString());
+        LogPrint(BCLog::VALIDATIONINTERFACE, "%s: new block hash=%s\n", func, pindexNew->GetBlockHash().ToString());
         m_internals->UpdatedBlockTip(pindexNew, pindexFork, fInitialDownload);
     });
 }
 
 void CMainSignals::TransactionAddedToMempool(const CTransactionRef &ptx) {
-    m_internals->m_schedulerClient.AddToProcessQueue([ptx, this] {
+    auto func = __func__;
+    m_internals->m_schedulerClient.AddToProcessQueue([func, ptx, this] {
+        LogPrint(BCLog::VALIDATIONINTERFACE, "%s: tx hash=%s\n", func, ptx->GetHash().ToString());
         m_internals->TransactionAddedToMempool(ptx);
     });
 }
 
 void CMainSignals::TransactionRemovedFromMempool(CTransactionRef ptx, MemPoolRemovalReason reason) {
     if (reason != MemPoolRemovalReason::BLOCK && reason != MemPoolRemovalReason::CONFLICT) {
-        m_internals->m_schedulerClient.AddToProcessQueue([ptx, this] {
+        auto func = __func__;
+        m_internals->m_schedulerClient.AddToProcessQueue([func, ptx, this] {
+            LogPrint(BCLog::VALIDATIONINTERFACE, "%s: tx hash=%s\n", func, ptx->GetHash().ToString());
```

Comment:
You may add MemPoolRemovalReasonString somewhere, like TransactionErrorString in util/error.cp. IMO that's an interesting info to log

Reply:
Done in 504d009.